k8s/autopilot/autopilotruleobject.go

```diff
@@ -0,0 +1,59 @@
+package autopilot
+
+import (
+	autv1alpha1 "github.com/libopenstorage/autopilot-api/pkg/apis/autopilot/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// RuleObjectOps is an interface to perform k8s AutopilotRuleObjects operations
+type RuleObjectOps interface {
+	// CreateAutopilotRuleObject creates the AutopilotRuleObject object
+	CreateAutopilotRuleObject(namespace string, object *autv1alpha1.AutopilotRuleObject) (*autv1alpha1.AutopilotRuleObject, error)
+	// GetAutopilotRuleObject gets the AutopilotRuleObject for the provided name
+	GetAutopilotRuleObject(namespace, name string) (*autv1alpha1.AutopilotRuleObject, error)
+	// UpdateAutopilotRuleObject updates the AutopilotRuleObject
+	UpdateAutopilotRuleObject(namespace string, object *autv1alpha1.AutopilotRuleObject) (*autv1alpha1.AutopilotRuleObject, error)
+	// DeleteAutopilotRuleObject deletes the AutopilotRuleObject of the given name
+	DeleteAutopilotRuleObject(namespace, name string) error
+	// ListAutopilotRules lists AutopilotRulesObjects
+	ListAutopilotRules(namespace string) (*autv1alpha1.AutopilotRuleObjectList, error)
+}
+
+// CreateAutopilotRuleObject creates the AutopilotRuleObject object
+func (c *Client) CreateAutopilotRuleObject(namespace string, ruleObject *autv1alpha1.AutopilotRuleObject) (*autv1alpha1.AutopilotRuleObject, error) {
+	if err := c.initClient(); err != nil {
+		return nil, err
+	}
+	return c.autopilot.AutopilotV1alpha1().AutopilotRuleObjects(namespace).Create(ruleObject)
+}
+// GetAutopilotRuleObject gets the AutopilotRuleObject for the provided name
+func (c *Client) GetAutopilotRuleObject(namespace,name string) (*autv1alpha1.AutopilotRuleObject, error) {
+	if err := c.initClient(); err != nil {
+		return nil, err
+	}
+	return c.autopilot.AutopilotV1alpha1().AutopilotRuleObjects(namespace).Get(name, metav1.GetOptions{})
+}
+
+// UpdateAutopilotRuleObject updates the AutopilotRuleObject
+func (c *Client) UpdateAutopilotRuleObject(namespace string, ruleObject *autv1alpha1.AutopilotRuleObject) (*autv1alpha1.AutopilotRuleObject, error) {
+	if err := c.initClient(); err != nil {
+		return nil, err
+	}
+	return c.autopilot.AutopilotV1alpha1().AutopilotRuleObjects(namespace).Update(ruleObject)
+}
+
+// DeleteAutopilotRuleObject deletes the AutopilotRuleObject of the given name
+func (c *Client) DeleteAutopilotRuleObject(namespace,name string) error {
```

Comment:
there is a go fmt tool in goland, you can check a checkbox and it will automatically run go fmt on every commit, otherwise this code will not pass the lint check